### PR TITLE
docs: fix code snippets for expo tutorial

### LINKED ISSUE #20764: Expo React Native tutorial has some TypeScript compiler errors
# Improve documentation

## Link 

[Build a User Management App with Expo React Native](https://supabase.com/docs/guides/getting-started/tutorials/with-expo-react-native#create-an-upload-widget)

## Describe the problem

1. The code of `components/Avatar.tsx` has some TypeScript errors. 
2. I also encountered [this issue](https://github.com/rnmods/react-native-document-picker/issues/677) and in my own demo I used `expo-document-picker` instead.

## Describe the improvement

1. Add a check and some type conversions.
2. I'm not sure if I should replace `react-native-document-picker` or just wait for the official fix, so I will separate this into another commit.

## Additional context

errors

### PR DESCRIPTION
**Pull request by NekodRider**
## What kind of change does this PR introduce?

Fix #20764 

## What is the new behavior?
1. No more errors from TypeScript compiler.
2. Replace `react-native-document-picker` with `expo-document-picker`.

### CODE REVIEW

File: apps/docs/pages/guides/getting-started/tutorials/with-expo-react-native.mdx [hunk 1]
```diff
@@ -454,7 +454,7 @@ photos and videos.
 You will need a file picker that works on the environment you will build the project for, we will use react-native-document-picker in this example.
```
Comment (charislam) [inline apps/docs/pages/guides/getting-started/tutorials/with-expo-react-native.mdx:454]:
```suggestion
You will need a file picker that works on the environment you will build the project for, we will use expo-document-picker in this example.
```

**Review by charislam — APPROVED** [review]

**Comment by bdotsamir** [comment]
This is great, however I've just submitted another PR that addresses a similar issue. The reason I submitted my PR was because I believe that a *document* picker is insufficient, unless the document picker can also include selecting from the user's camera roll.

#20796

**Comment by NekodRider** [comment]
> This is great, however I've just submitted another PR that addresses a similar issue. The reason I submitted my PR was because I believe that a _document_ picker is insufficient, unless the document picker can also include selecting from the user's camera roll.
> 
> #20796

Yeah, I checked your PR and agree with that using `expo-image-picker` is more suitable for this tutorial. But this PR also includes a stylesheet fix for TypeScript compiler.

**Comment by bdotsamir** [comment]
Got it, would you then like to move my changes into this PR?

Technically you were in this space first, I don't want to steal your PR lol

**Comment by NekodRider** [comment]
> Got it, would you then like to move my changes into this PR?
> 
> Technically you were in this space first, I don't want to steal your PR lol

:) Thank you. I merged your branch into mine and added some changes mentioned in your PR.

**Review by charislam — APPROVED** [review]

**Comment by charislam** [comment]
Thanks for the collab! ❤️

**Pull request merged**